charts/telepresence-oss/values.schema.yaml

```
@@ -0,0 +1,580 @@
+        description: Set this port number to enable a prometheus metrics http server for the traffic manager
+        type: integer
+
+  rbac:
+    type: object
+    additionalProperties: false
+    properties:
+      only:
+        description: Only create the RBAC resources and omit the traffic-manger
+        type: boolean
+
+  readinessProbe:
+    description: Define readinessProbe for the Traffic Manger.
+    $ref: "#/$defs/probe"
+
+  replicaCount:
+    description: Number or replicas for the traffic-manager. The Traffic Manager only support running with one replica at the moment.
+    type: integer
+    minimum: 1
+    maximum: 1
+
+  resources:
+    description: Define resource requests and limits for the Traffic Manger
+    $ref: "#/$defs/resourceRequirements"
+
+  schedulerName:
+    description: Specify a scheduler for Traffic Manager Pod and hooks Pod
+    type: string
+
+  securityContext:
+    description: The Kubernetes SecurityContext for the traffic-manager container
+    $ref: "#/$defs/securityContext"
+
+  service:
+    type: object
+    additionalProperties: false
+    properties:
+      type:
+        description: The type of service for the traffic-manager
+        type: string
+
+  telepresenceAPI:
+    type: object
+    additionalProperties: false
+    properties:
+      port:
+        description: The port on agent's localhost where the Telepresence API server can be found
+        type: integer
+
+  timeouts:
+    type: object
+    additionalProperties: false
+    properties:
+      agentArrival:
+        description: The time that the traffic-manager will wait for the traffic-agent to arrive
+        $ref: "#/$defs/duration"
+
+  tolerations:
+    description: Define tolerations for the Traffic Manager to ignore Node taints
+    type: array
+    items:
+      $ref: "#/$defs/toleration"
+
+  workloads:
+    type: object
+    additionalProperties: false
+    properties:
+      argoRollouts:
+        type: object
+        additionalProperties: false
+        properties:
+          enabled:
+            description: Enable/Disable the argo-rollouts integration
+            type: boolean
+      deployments:
+        type: object
+        additionalProperties: false
+        properties:
+          enabled:
+            description: Enable/Disable the support for Deployments
+            type: boolean
+      replicaSets:
+        type: object
+        additionalProperties: false
+        properties:
+          enabled:
+            description: Enable/Disable the support for ReplicaSets
+            type: boolean
+      statefulSets:
+        type: object
+        additionalProperties: false
+        properties:
+          enabled:
+            description: Enable/Disable the support for StatefulSets
+            type: boolean
+
+$defs:
+  duration:
+    pattern: "^[+-]?(\\d+(\\.\\d+)?(h|m|s|ms|us|µs))+"
+  rfc1123Label:
+    description: RFC 1123 label name
+    type: string
+    pattern: "^[a-z0-9][a-z0-9-]{1,62}$"
```

I believe this regex isn't fully valid for rfc1123, as IIRC the label must end with an alphanumeric character, so it should be

```suggestion
    pattern: "^[a-z0-9](?:[a-z0-9-]{,61}[a-z0-9])?$"
```
this should be more valid

---

Agree, but why use a non-capturing group? We don't use the result anyway so it just makes the regexp harder to read.